### PR TITLE
feat: experiment on removing profile picture during onboarding flow

### LINKED ISSUE #2112: Experiment: Remove profile picture from onboarding flow
[Placeholder for experiment proposal here]
There's a ~25.6% dropoff from setting name and picture page during the onboarding flow.  The hypothesis is it's unclear from our UI that profile picture is optional so user drop off from this page because there're no pictures they have at hand.

AC:
- there is a feature flag and A/B test set up in firebase according to specs from the proposal
- profile picture during the onboarding flow is hidden for the "enabled" group

### PR DESCRIPTION
**Pull request by lisaqyli**
### Description

[Experiment Proposal]( https://www.notion.so/valora-inc/Experiment-Design-Profile-picture-removal-in-onboarding-844eba5dfd70481495c3b750a59e9ed5)

The PR also fixes the event property `includesPhoto` for `name_and_picture_set` event (independent from the experiment)

Adding a remote config to remove profile picture from new account creation step 1
274901369_494672122186732_568536932096933050_n

### Tested

unit tested

### How others should test
On alfajores, create a new account, profile picture input should be hidden on step 1 of new account creation

### Related issues

- Fixes #2112

### CODE REVIEW

File: packages/mobile/src/analytics/Properties.tsx [hunk 1]
```diff
@@ -225,6 +225,7 @@ interface OnboardingEventsProperties {
 
   [OnboardingEvents.name_and_picture_set]: {
     includesPhoto: boolean
+    profile_picture_skipped: boolean
```
Comment (cajubelt) [inline packages/mobile/src/analytics/Properties.tsx:228]: why snake case here? looks like the other fields are camelCase

File: packages/mobile/src/onboarding/registration/NameAndPicture.tsx [hunk 2]
```diff
@@ -79,7 +81,8 @@ function NameAndPicture({ navigation }: Props) {
 
     dispatch(setPromptForno(true)) // Allow forno prompt after Welcome screen
     ValoraAnalytics.track(OnboardingEvents.name_and_picture_set, {
-      includesPhoto: false,
+      includesPhoto: photoChosen,
```
Comment (cajubelt) [inline packages/mobile/src/onboarding/registration/NameAndPicture.tsx:84]: this was just hardcoded to false before? weird..

**Review by cajubelt — APPROVED** [review]
🥳 

left a minor thing about casing

**Review by jh2oman — CHANGES_REQUESTED** [review]
PR looks good, but needs a test for the migrations.ts change. We've gotten burned a few times in the past few weeks by not having tests there.

File: packages/mobile/src/redux/migrations.ts [hunk 3]
```diff
@@ -474,4 +474,11 @@ export const migrations = {
       paymentDeepLinkHandler: PaymentDeepLinkHandler.Disabled,
     },
   }),
+  40: (state: any) => ({
+    ...state,
+    app: {
+      ...state.app,
+      skipProfilePicture: false,
```
[the remark this answers is not in the document]
Comment (lisaqyli) [reply packages/mobile/src/redux/migrations.ts:481]: no particular reason, using the default value might make it slightly easier to read.

File: packages/mobile/src/onboarding/registration/NameAndPicture.tsx (hunk 2 above)
Comment (lisaqyli) [reply packages/mobile/src/onboarding/registration/NameAndPicture.tsx:84]: i know...

File: packages/mobile/src/onboarding/registration/NameAndPicture.tsx [hunk 4]
```diff
@@ -80,6 +81,7 @@ function NameAndPicture({ navigation }: Props) {
     dispatch(setPromptForno(true)) // Allow forno prompt after Welcome screen
     ValoraAnalytics.track(OnboardingEvents.name_and_picture_set, {
       includesPhoto: false,
```
Comment (jh2oman) [inline packages/mobile/src/onboarding/registration/NameAndPicture.tsx:83]: Its a bit confusing leaving this property around. Do we still need it?

**Review by jh2oman — APPROVED** [review]
Looks great!

File: packages/mobile/src/onboarding/registration/NameAndPicture.tsx (hunk 4 above)
Comment (lisaqyli) [reply packages/mobile/src/onboarding/registration/NameAndPicture.tsx:83]: i think this prop means if whether a profile picture is included. The default `false` is incorrect. I had a fix for it but reverted that b/c it was difficult to unit test pictureInput component. I'm planning on fixing it in a separate PR

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified above task on Latest Test Flight Alfa jores build V1.29.0, Android Alfa jores Build V1.29.0 & updated status in test rails. On "step 1" while creating and restoring account only name field and next button appears and app photo is not appearing and user is able to create and restore account successfully

**Test rail link:**
**Android 11:** https://valoraapp.testrail.io/index.php?/tests/view/20918
**iOS13:** https://valoraapp.testrail.io/index.php?/tests/view/20855
**iOS14:** https://valoraapp.testrail.io/index.php?/tests/view/20792

**Devices:** Google Pixel 2XL (11.0), iPhone SE 2020 (14.8.1), iPhone 7 (13.3)
**cc:** @lisaqyli  
Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified above task on Latest Android Alfa jores Build V1.29.2 & updated status in test rails. 

- On "step 1" while creating and restoring account only name field and next button appears and app photo is not appearing and user is able to create and restore account successfully.

**Test rail link:**
**Android 12:** https://valoraapp.testrail.io/index.php?/tests/view/20981
**Device:** Google Pixel 4a (12.0)
CC: @lisaqyli 
Thanks..!!

**Pull request merged**